File: Kanbardoo.Domain/Entities/KanTask.cs
using Kanbardoo.Domain.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Kanbardoo.Domain.Entities;

public class KanTask : Entity
{
    [MaxLength(256)]
    public string Name { get; set; } = string.Empty;
    [MaxLength(1024)]
    public string Description { get; set; } = string.Empty;
    public DateTime DueDate { get; set; }
    public int StatusID { get; set; }

    [ForeignKey(nameof(StatusID))]
    public KanTaskStatus Status { get; set; } = new KanTaskStatus();
    public int AssigneeID { get; set; }

    [ForeignKey(nameof(AssigneeID))]
    public KanUser Assignee { get; set; } = new KanUser();
    public int TableID { get; set; }
    [ForeignKey(nameof(TableID))]
    public KanTable Table { get; set; } = new KanTable();
}

File: frontend/src/app/_models/kan-task.ts
import { KanTaskStatus } from "./kan-task-status";
import { KanUser } from "./kan-user";

export interface KanTask {
    id: number;
    name: string;
    description: string;
    dueDate: Date;
    status: KanTaskStatus;
    assignee: KanUser;
    tableID: number;
    priority: number;
}

export function getDefaultTask() {
    return {
        id: 0,
        name: '',
        description: '',
        dueDate: new Date(),
        status: { id: 0, name: '' },
        assignee: { id: 0, userName: '', creationDate: new Date() },
        tableID: 0,
        priority: 0
    };
}